dev/com.ibm.ws.security/src/com/ibm/ws/security/AccessIdUtil.java

```diff
@@ -190,18 +202,65 @@ public static String getUniqueId(String accessId) {
     public static String getUniqueId(String accessId, String realm) {
     
         if (realm != null) {
-            Pattern pattern = Pattern.compile("([^:]+):(" + Pattern.quote(realm) + ")/(.*)");
+            RealmHolder holder = realmHolder;
+            Pattern pattern = null;
+            if (holder != null && holder.realmPattern != null && realm.equals(holder.realm[0])) {
+                pattern = holder.realmPattern;
+            } else {
+                pattern = Pattern.compile("([^:]+):(" + Pattern.quote(realm) + ")/(.*)");
+            }
```

Comment:
Could be put in a method so it can be reused here and in `getUniqueIds()`.

Reply:
Done